packages/access-api/src/durable-objects/space-verifier.js

```diff
@@ -0,0 +1,131 @@
+import { stringToDelegation } from '@web3-storage/access/encoding'
+
+/**
+ *
+ * @template {import('@ucanto/interface').Capabilities} [T=import('@ucanto/interface').Capabilities]
+ * @param {import('../bindings').DurableObjectNamespace} spaceVerifiers
+ * @param {string} space
+ * @param {import('@web3-storage/access/src/types').EncodedDelegation<T>} ucan
+ */
+export async function sendDelegationToSpaceVerifier(
+  spaceVerifiers,
+  space,
+  ucan
+) {
+  const durableObjectID = spaceVerifiers.idFromName(space)
+  const durableObject = spaceVerifiers.get(durableObjectID)
+  // hostname is totally ignored by the durable object but must be set so set it to example.com
+  const response = await durableObject.fetch('https://example.com/delegation', {
+    method: 'PUT',
+    body: ucan,
+  })
+  if (response.status === 400) {
+    throw new Error(response.statusText)
+  }
+}
+
+/**
+ * @template {import('@ucanto/interface').Capabilities} [T=import('@ucanto/interface').Capabilities]
+ * @param {WebSocket} server
+ * @param {import('@web3-storage/access/src/types').EncodedDelegation<T>} ucan
+ */
+function sendDelegation(server, ucan) {
+  server.send(
+    JSON.stringify({
+      type: 'delegation',
+      delegation: ucan,
+    })
+  )
+  server.close()
+}
+
+/**
+ * SpaceVerifier
+ */
+export class SpaceVerifier {
+  /**
+   * @param {import('../bindings').DurableObjectState} state
+   */
+  constructor(state) {
+    this.state = state
+    // `blockConcurrencyWhile()` ensures no requests are delivered until
+    // initialization completes.
+    this.state.blockConcurrencyWhile(async () => {
+      this.ucan = await this.state.storage.get('ucan')
+    })
+  }
+
+  cleanupServer() {
+    this.server = undefined
+  }
+
+  async cleanupUCAN() {
+    this.ucan = undefined
+    await this.state.storage.put('ucan', '')
+  }
+
+  /**
+   * @param {Request} req
+   */
+  async fetch(req) {
```

Comment:
I would expect to see an authorization check somewhere to make we don't give the ucan to someone who shouldn't have it?

If someone else tries to connects first before the client we expect, what would happen? (It's been a long day, but I read it like that would deny service to the desired user)

I wonder if (long term) maybe this should be like `AgentVerifier` and be keyed off of the agent did, not space did.
(or `agent did` + `cid(authorizationRequest)` even).

It seems like keying this off of space and only allowing one server/ucan at a time will:
* prevent more than one of these flows happening at a time for a space - not really needed for `voucher/redeem` but it makes this less useful with new `authorize` flow

Reply:
> I would expect to see an authorization check somewhere to make we don't give the ucan to someone who shouldn't have it?

hm, I would probably handle this further up the stack, but also I'm not sure we do this with the existing implementation and if we do I think we're still doing that for this flow? 

tbh I'm not entirely sure whether we need to do additional auth here or not, but I'm not sure what that would look like concretely - do you have some sort of signature verification in mind?

Reply:
> I wonder if (long term) maybe this should be like AgentVerifier and be keyed off of the agent did, not space did.
> (or agent did + cid(authorizationRequest) even).

I think this is exactly the direction we should go with the new stuff coming in yep

Reply:
> It seems like keying this off of space and only allowing one server/ucan at a time will prevent more than one of these flows happening at a time for a space - not really a problem for voucher but it makes this less useful with new access/authorize + access/claim flow which is not 1:1 with a single space.

yep - keying off Agent will be the way to go for the path that handles the `access/authorize` flow

Reply:
I think invocation CID might be better yet as agent in theory might start multiple authorization flows or maybe we'll want to deliver other messages over websocket.

Reply:
hm I don't totally follow, but seems reasonable as long as both sides of the equation (ie, the code calling registerSpace and the code handling the request from the email link) can both calculate the invocation CID